### PR TITLE
Adds more specific regex validation to UPS connect form

### LINKED ISSUE #2136: Connecting a UPS account, in-line form field errors
Noticed some things as I was testing the form field validations that would help people complete the form easier. The current problem is the message, "This field is required" doesn't provide enough information in case they've entered something incorrectly.

For the UPS account number error message (the first field), if the user enters less than 6 digits in the field the error message could say
> Your UPS account number needs to be 6 digits in length

For the Zip code fields, (since these are all US-based users) for anything that's not 5 digits we could also show the error message as:
> The zip code needs to be 5 digits in length 

For the phone number field, for anything less than 10 digits:
> The phone number needs to be 10 digits in length

Alternatively, if we can't get that specific we could use the same error state message as the current one for the website:
> The [form field] format is not valid

image

These would appear only after the user has interacted with the field.

### PR DESCRIPTION
**Pull request by FangedParakeet**
## Description

The current form field validations when a user is connecting their UPS account don't always provide enough information to inform a user on how to correct them: in a few cases, the only validation present was checking non-empty values for required fields and in others the error lacked specificity. 

This PR adds the following validations for fields in this form.

- **UPS account number**: `The UPS account number needs to be 6 digits in length`
- **Phone number**: `The phone number needs to be 10 digits in length`
- **ZIP/Postal Code** (US only): `The ZIP/Postal code needs to be 5 digits in length`
- **UPS Invoice Number**: If present, `The invoice number needs to be 9 or 13 digits in length`
- **UPS Invoice Date**: If present, `The date must be a valid date in the format YYYY-MM-DD`

### Related issue(s)
Closes #2136 

### Steps to reproduce & screenshots/GIFs

Enable UPS for your site, by adding a row to `wcc_site_flags` with `flag_name` as `can_use_ups` and your site's ID for `siteid`. Refresh the WCS plugin from **WooCommerce > Status > WooCommerce Shipping & Tax** to complete the enablement.
Enable UPS beta

Go to **WooCommerce > Settings > Shipping > WooCommerce Shipping** and from the Carrier Account section at the bottom of the page, select "Connect" next to UPS.
Connect UPS account

Enter values into the fields mentioned above and check the validation.
UPS account form field

### Checklist

- [ ] unit tests

- [x] `changelog.txt` entry added

### CODE REVIEW

File: client/extensions/woocommerce/woocommerce-services/state/carrier-accounts/selectors.js [hunk 1]
```diff
@@ -39,17 +43,38 @@ export const getFormErrors = ( state, siteId, carrier ) => {
 		}
 	}
 
+	if ( values.account_number && ! values.account_number.match( UPSAccountNumberRegex ) ) {
+		fieldErrors.account_number = translate( 'The UPS account number needs to be 6 digits in length' );
```
Comment (frosso) [inline client/extensions/woocommerce/woocommerce-services/state/carrier-accounts/selectors.js:47]: The account number can be digits or letters, its length should be max 6 characters.
This is validating only on the digits, it should also include `[a-zA-Z]`

Comment (FangedParakeet) [reply client/extensions/woocommerce/woocommerce-services/state/carrier-accounts/selectors.js]: Got it, @frosso, I didn't realise this! I was following @elizaan36's note on the linked issue blindly, like a giant worm chasing Kevin Bacon off the edge of a cliff. I've just updated this PR with a new commit that corrects this. You should now be able to see the following error messages: 

Error state
_Error state_

Letters accepted
_Letters accepted_

```
const UPSAccountNumberRegex = /^[a-zA-Z0-9]{6}$/;
```
_New regex_

Also, I've today realised that my GitHub notifications are basically all set wrong, so sorry @frosso for taking so long to notice your feedback! 🤦

Comment (frosso) [reply client/extensions/woocommerce/woocommerce-services/state/carrier-accounts/selectors.js]: haha no worries! :) The new regex looks good!

**Review by frosso — APPROVED** [review]
Looks good to me! 👍

**Pull request merged**